### PR TITLE
combined user endpoints and defined add user mutation

### PR DESCRIPTION
**Pull request by steven-ha**
simple query and mutation for user

### CODE REVIEW

File: schema/schema.js [hunk 1]
```diff
@@ -278,16 +279,60 @@ const RootQuery = new GraphQLObjectType({
...
+      type: UserType,
+      args: {
+        id: { type: GraphQLInt },
+        first_name: { type: GraphQLString },
+        last_name: { type: GraphQLString },
+        email: { type: GraphQLString },
+        phone: { type: GraphQLString },
+      },
+      resolve(parent, args) {
+        let newUser = {
```
Comment (mybrainishuge) [inline schema/schema.js]: This should be `const`, but see my comment below before changing this. I'm not seeing the reason for storing it in a variable.

File: schema/schema.js [hunk 2]
```diff
@@ -278,16 +279,60 @@ const RootQuery = new GraphQLObjectType({
...
+          phone: args.phone,
+          address1: "4529 W 34th Ave",
+          address2: "",
+          city: "Denver",
+          state: 1,
+          country: 1,
+          zipcode: "80212",
+          height: 5.7,
+          weight: 146,
+          measurement_system: 1,
```
Comment (mybrainishuge) [inline schema/schema.js]: I assume most of these values are hardcoded now just to simplify building out the initial view, yeah?

File: schema/schema.js [hunk 3]
```diff
@@ -278,16 +279,60 @@ const RootQuery = new GraphQLObjectType({
...
+        };
+
+        user.push(newUser);
+
+        return newUser;
```
Comment (mybrainishuge) [inline schema/schema.js]: I'm not seeing where `user` is defined. Are you pushing into an undefined user array? Can this be simplified to return the new user object literal?

File: schema/schema.js [hunk 4]
```diff
@@ -278,16 +279,60 @@ const RootQuery = new GraphQLObjectType({
   name: "RootQueryType",
   fields: {
     user: {
-      type: UserType,
+      type: GraphQLList(UserType),
       args: { id: { type: GraphQLInt } },
```
Comment (mybrainishuge) [inline schema/schema.js]: I thought you removed the args `id` in order to make it optional. What am I missing?

File: data/data.js [hunk 5]
```diff
@@ -18,6 +18,25 @@ const user = [
     weight: 175,
     measurement_system: 1,
   },
+  {
+    id: 2,
+    user_type: 1,
+    first_name: "Steven",
+    last_name: "Ha",
+    age: 36,
```
Comment (mybrainishuge) [inline data/data.js]: You ain't 36.

Comment (steven-ha) [reply data/data.js]: People have said I look young for my age, I'm tired of it.

File: schema/schema.js (hunk 4 above)
Comment (steven-ha) [reply schema/schema.js]: Graphql types are nullable by default. When we query the graphql server, we could omit those arguments we don't need. If an argument is required, we'd specify the type as GraphQLNonNull<type>.

File: schema/schema.js (hunk 3 above)
Comment (steven-ha) [reply schema/schema.js]: We're using arrays to represent our database at the moment. When we execute our user mutation, it's adding the user object to that array so we have access to that user later on. The user array is defined in the 'data.js' file.

File: schema/schema.js (hunk 2 above)
Comment (steven-ha) [reply schema/schema.js]: Yea, I was to lazy when I got around to defining the arguments for the adduser method.

File: schema/schema.js (hunk 4 above)
Comment (mybrainishuge) [reply schema/schema.js]: Then why were you getting an error when you removed the argument earlier?

File: schema/schema.js (hunk 3 above)
Comment (mybrainishuge) [reply schema/schema.js]: Ohhh right. So this `user.push` business will be replaced with a db update later.

File: schema/schema.js (hunk 1 above)
Comment (steven-ha) [reply schema/schema.js]: I agree, should be a constant. Doesn't have to be stored, I could return the last item in the array after pushing a user object to the user array.

Comment (mybrainishuge) [reply schema/schema.js]: Nah, this is fine. Just change it to a `const`. Returning the last item would look like `user[user.length - 1]` which looks dumb to me.

File: schema/schema.js (hunk 4 above)
Comment (steven-ha) [reply schema/schema.js]: The error was from having empty parentheses after the query name.

**Pull request merged**